bson/src/main/scala/types.scala

```diff
@@ -492,15 +442,12 @@ object BSONObjectID {
     id(3) = (timestamp & 0xFF).toByte
 
     if (!fillOnlyTimestamp) {
-      // machine id, 3 first bytes of md5(macadress or hostname)
-      id(4) = machineId(0)
-      id(5) = machineId(1)
-      id(6) = machineId(2)
-
-      // 2 bytes of the pid or thread id. Thread id in our case. Low endian
-      val threadId = Thread.currentThread.getId.toInt
-      id(7) = (threadId & 0xFF).toByte
-      id(8) = (threadId >> 8 & 0xFF).toByte
+      id(4) = (randomVal1 >> 16 & 0xFF).toByte
+      id(5) = (randomVal1 >> 8 & 0xFF).toByte
+      id(6) = (randomVal1 & 0xFF).toByte
```

Comment:
Wouldn't it be more straightforward to have 5 bytes variables initialized already in constructor? I also do not get `randomVal1` and `randomVal2` - wouldn't it be easier to throw random long and then pick 5 bytes from it?

Reply:
That was directly from the Java drivers, but we don't need to expose the maching/process id part which they did. So just replaced it with nextBytes to return 5 random bytes.